core/src/gateway_internal.c

```diff
@@ -690,23 +690,26 @@ int add_module_to_any_source(GATEWAY_HANDLE_DATA* gateway_handle, MODULE_DATA* m
 
 void remove_module_from_any_source(GATEWAY_HANDLE_DATA* gateway_handle, MODULE_DATA* module)
 {
-    size_t link;
-    size_t num_links = VECTOR_size(gateway_handle->links);
-    for (link = 0; link < num_links; link++)
+    if (gateway_handle->links)
     {
-        LINK_DATA * link_data = VECTOR_element(gateway_handle->links, link);
-        if (link_data->from_any_source)
+        size_t link;
+        size_t num_links = VECTOR_size(gateway_handle->links);
+        for (link = 0; link < num_links; link++)
         {
-            MODULE_DATA** module_sink = (MODULE_DATA**)VECTOR_find_if(gateway_handle->modules, module_name_find, link_data->module_sink->module_name);
-            if (module_sink == NULL)
+            LINK_DATA * link_data = VECTOR_element(gateway_handle->links, link);
+            if (link_data->from_any_source)
             {
-                LogError("Could not find sink for link [%s]", link_data->module_sink);
-            }
-            else
-            {
-                if (remove_one_link_from_broker(gateway_handle, module->module, (*module_sink)->module) != 0)
+                MODULE_DATA** module_sink = (MODULE_DATA**)VECTOR_find_if(gateway_handle->modules, module_name_find, link_data->module_sink->module_name);
```

Comment:
I don't think so. If `gateway_handle->links` is non-NULL, and VECTOR_size returns a non-zero value, then the vector element will be valid. We maintain this vector internally, so we can guarantee consistency/integrity.